testsuite/integration/elytron/src/test/java/org/wildfly/test/integration/elytron/securityapi/SecurityAPITestCase.java

```diff
@@ -0,0 +1,223 @@
+import static org.wildfly.test.integration.elytron.securityapi.TestAuthenticationMechanism.MESSAGE_HEADER;
+import static org.wildfly.test.integration.elytron.securityapi.TestAuthenticationMechanism.PASSWORD_HEADER;
+import static org.wildfly.test.integration.elytron.securityapi.TestAuthenticationMechanism.USERNAME_HEADER;
+import static org.wildfly.test.integration.elytron.securityapi.TestIdentityStore.PASSWORD;
+import static org.wildfly.test.integration.elytron.securityapi.TestIdentityStore.USERNAME;
+
+import java.net.URI;
+import java.net.URL;
+
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.util.EntityUtils;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.arquillian.test.api.ArquillianResource;
+import org.jboss.as.arquillian.api.ServerSetup;
+import org.jboss.as.controller.PathAddress;
+import org.jboss.as.controller.PathElement;
+import org.jboss.as.controller.client.ModelControllerClient;
+import org.jboss.as.controller.operations.common.Util;
+import org.jboss.as.test.integration.management.util.CLIWrapper;
+import org.jboss.as.test.integration.security.common.Utils;
+import org.jboss.dmr.ModelNode;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.wildfly.test.security.common.AbstractElytronSetupTask;
+import org.wildfly.test.security.common.elytron.ConfigurableElement;
+import org.wildfly.test.undertow.common.UndertowApplicationSecurityDomain;
+
+/**
+ * Test case to test the EE Security API with WildFly Elytron.
+ *
+ * @author <a href="mailto:darran.lofthouse@jboss.com">Darran Lofthouse</a>
+ */
+@RunWith(Arquillian.class)
+@RunAsClient
+@ServerSetup({ SecurityAPITestCase.ServerSetup.class })
+public class SecurityAPITestCase {
+
+    @ArquillianResource
+    protected URL url;
+
+    @Deployment
+    protected static WebArchive createDeployment() {
+        final Package testPackage = SecurityAPITestCase.class.getPackage();
+        return ShrinkWrap.create(WebArchive.class, SecurityAPITestCase.class.getSimpleName() + ".war")
+                .addClasses(SecurityAPITestCase.class, AbstractElytronSetupTask.class, ServerSetup.class)
+                .addClasses(TestAuthenticationMechanism.class, TestIdentityStore.class)
+                .addClasses(TestServlet.class)
+                .addClasses(WhoAmI.class, WhoAmIBean.class)
+                .addAsWebInfResource(Utils.getJBossWebXmlAsset("SecurityAPI"), "jboss-web.xml")
+                .addAsWebInfResource(testPackage, "jboss-ejb3.xml", "jboss-ejb3.xml")
+                .addAsWebInfResource(testPackage, "beans.xml", "beans.xml");
```

Comment:
For execution with security manager, something like the following is necessary:
`.addAsManifestResource(createPermissionsXmlAsset(new ElytronPermission("getSecurityDomain")), "permissions.xml")`

Reply:
Having the permission set, the test passes. However, there are subsequent errors.

[2018-12-19T11-33-25_020-jvmRun1.dump.txt](https://github.com/wildfly/wildfly/files/2694449/2018-12-19T11-33-25_020-jvmRun1.dump.txt)
[org.wildfly.test.integration.elytron.securityapi.SecurityAPITestCase-output.txt](https://github.com/wildfly/wildfly/files/2694450/org.wildfly.test.integration.elytron.securityapi.SecurityAPITestCase-output.txt)

Reply:
Is passing with a security manager a requirement?

Reply:
Regarding the test, I can't find it written anywhere, but the tests in the WildFly test suite should pass with security manager. The same for a new functionality.
The test output contains unexpected errors, like
`
ERROR [org.jboss.msc.service.fail] (MSC service thread 1-4) MSC000001: Failed to start service jboss.as: org.jboss.msc.service.StartException in service jboss.as: Failed to start service
	at org.jboss.msc.service.ServiceControllerImpl$StartTask.execute(ServiceControllerImpl.java:1730)
	at org.jboss.msc.service.ServiceControllerImpl$ControllerTask.run(ServiceControllerImpl.java:1558)
	at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
	at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:1985)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1487)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1378)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.security.AccessControlException: WFSM000001: Permission check failed (permission "("java.lang.RuntimePermission" "canCreateModuleLoader")" in code source "(file:/home/okotek/test/wfly7186/testsuite/integration/elytron/target/wildfly/jboss-modules.jar <no signer certificates>)" of "sun.misc.Launcher$AppClassLoader@70dea4e")
	at org.wildfly.security.manager.WildFlySecurityManager.checkPermission(WildFlySecurityManager.java:294)
	at org.wildfly.security.manager.WildFlySecurityManager.checkPermission(WildFlySecurityManager.java:191)
	at org.jboss.modules.ModuleLoader.checkPermissions(ModuleLoader.java:204)
	at org.jboss.modules.ModuleLoader.<init>(ModuleLoader.java:174)
	at org.jboss.modules.ModuleLoader.<init>(ModuleLoader.java:165)
	at org.jboss.as.server.moduleservice.ServiceModuleLoader.<init>(ServiceModuleLoader.java:133)
	at org.jboss.as.server.moduleservice.ServiceModuleLoader.addService(ServiceModuleLoader.java:203)
	at org.jboss.as.server.ApplicationServerService.start(ApplicationServerService.java:156)
	at org.jboss.msc.service.ServiceControllerImpl$StartTask.startService(ServiceControllerImpl.java:1738)
	at org.jboss.msc.service.ServiceControllerImpl$StartTask.execute(ServiceControllerImpl.java:1700)
	... 6 more
`

It is not obvious whether this is a test issue or whether there is something really bad in the RFE functionality.

Reply:
@bstansberry do you have any info on the requirements to run with a security manager as we are currently not doing this in CI?

This functionality is extensively tested with a security manager elsewhere.

Reply:
The "Linux with security manager" PR job looks to be trying to do a general run with secmgr in the args to launch the server (or domain server). That's what the profile turned on by -Dsecurity.manager does.  That seems like a general thing, not something targeted at a subset of tests. And I know we get bug reports and eventually fixes when tests don't pass with security manager enabled. So since we're doing that we should make the tests pass in the first place.